### ISSUE TITLE
Flushing stdout with a binary protocol does not work

### ISSUE THREAD

**Issue by jothan** [tokio-rs/tokio#1527]
I'm having trouble with a binary protocol that does not use newlines as line separators. stdout seems to flush immediately only when there is a newline in the output.

I have included a snippet to reproduce this issue.

Flushing the underlying std stream works, but only when the async stream was flushed first.

stderr has the expected behaviour of flushing immediately correctly.

## Version
0.2.0-alpha.4

## Platform
Linux 4.19.0-5-amd64 #1 SMP Debian 4.19.37-5+deb10u2 (2019-08-08) x86_64 GNU/Linux

## Description

```rust
use std::io::Write;
use std::time::Duration;

use tokio::prelude::*;
use tokio::timer::delay;
use tokio::io::{stdout};

async fn works() -> Result<(), Box<dyn std::error::Error>> {
    let mut output = stdout();
    output.write_all(b"hello").await?;
    output.flush().await?;
    std::io::stdout().flush()?;

    Ok(())
}

async fn should_work() -> Result<(), Box<dyn std::error::Error>> {
    let mut output = stdout();
    output.write_all(b"hello").await?;
    output.flush().await?;

    Ok(())
}

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Both calls should output "hello" without a newline immediately
    //works().await?;
    should_work().await?;

    let when = tokio::clock::now() + Duration::from_secs(5);
    delay(when).await;
    
    Ok(())
}
```

**Comment by jothan** [comment]
Adding inner.flush() in Blocking::poll_flush() seems to fix the behaviour. I am not sure where to call the lower layer flush.

**Comment by carllerche** [comment]
Fixed by #1528
